packages/terra-date-picker/lib/DatePicker.js

```diff
@@ -0,0 +1,231 @@
+   */
+  includeDates: _react.PropTypes.arrayOf(_react.PropTypes.object),
+  /**
+   * A moment object to represents the maximum date that can be selected.
+   */
+  maxDate: _react.PropTypes.object,
+  /**
+   * A moment object to represents the minimum date that can be selected.
+   */
+  minDate: _react.PropTypes.object,
+  /**
+   * The selected date to show in the date input.
+   */
+  selectedDate: _react.PropTypes.object,
+  /**
+   * Indicates the end date picker of a date range.
+   */
+  isEndDateRange: _react.PropTypes.bool,
+  /**
+   * Indicates the start date picker of a date range.
+   */
+  isStartDateRange: _react.PropTypes.bool,
+  /**
+   * The default start date for a date range.
+   */
+  startDate: _react.PropTypes.object,
+  /**
+   * Indicates to display the picker in a full screen overlay.
+   */
+  withPortal: _react.PropTypes.bool,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   */
+  onChange: _react.PropTypes.func
+};
+
+var defaultProps = {
+  isEndDateRange: false,
+  isStartDateRange: false,
+  withPortal: undefined
+};
+
+var DatePicker = function (_React$Component) {
+  _inherits(DatePicker, _React$Component);
+
+  function DatePicker(props) {
+    _classCallCheck(this, DatePicker);
+
+    var _this = _possibleConstructorReturn(this, (DatePicker.__proto__ || Object.getPrototypeOf(DatePicker)).call(this, props));
+
+    _this.state = {
+      startDate: props.startDate,
+      endDate: props.endDate
+    };
+    _this.handleChange = _this.handleChange.bind(_this);
+    return _this;
+  }
+
+  _createClass(DatePicker, [{
+    key: 'handleChange',
+    value: function handleChange(date) {
+      this.setState({
+        startDate: date
+      });
+
+      if (this.props.onChange) {
+        this.props.onChange(date);
+      }
+    }
+  }, {
+    key: 'render',
+    value: function render() {
+      var _props = this.props,
+          customInput = _props.customInput,
+          endDate = _props.endDate,
+          excludeDates = _props.excludeDates,
+          filterDate = _props.filterDate,
+          includeDates = _props.includeDates,
+          maxDate = _props.maxDate,
+          minDate = _props.minDate,
+          isEndDateRange = _props.isEndDateRange,
+          isStartDateRange = _props.isStartDateRange,
+          selectedDate = _props.selectedDate,
+          startDate = _props.startDate,
+          withPortal = _props.withPortal,
+          customProps = _objectWithoutProperties(_props, ['customInput', 'endDate', 'excludeDates', 'filterDate', 'includeDates', 'maxDate', 'minDate', 'isEndDateRange', 'isStartDateRange', 'selectedDate', 'startDate', 'withPortal']);
+
+      var classes = (0, _classnames2.default)(['terra-DatePicker']);
+
+      // TODO: Need translation from date_util
+      var todayString = 'Today';
+
+      // TODO: Get the locale from date_util
+      var userLocale = window.navigator.language;
```

Comment:
We should probably discuss how locale should be set for an application. 

We need to ensure react-intl and momentjs are always on the same locale.